S/Sundials/build_tarballs.jl

```diff
@@ -1,31 +1,32 @@
 using BinaryBuilder
 
 name = "Sundials"
-version = v"3.1.1"
+version = v"5.1.0"
 
 # Collection of sources required to build SundialsBuilder
 sources = [
     "https://github.com/LLNL/sundials/archive/v$(version).tar.gz" =>
-    "d03fca24d7adbfe52e05bd4e61340251e1cf56927540cc77a1ca817716091166",
+    "101be83221f9a0ab185ecce04d003ba38660cc71eb81b8a7cf96d1cc08b3d7f9",
     "./bundled",
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
 cd $WORKSPACE/srcdir/sundials-*/
-patch -p0 < $WORKSPACE/srcdir/patches/Sundials_windows.patch
+# patch -p0 < $WORKSPACE/srcdir/patches/Sundials_windows.patch
 
 CMAKE_FLAGS=(-DCMAKE_INSTALL_PREFIX=${prefix} -DCMAKE_TOOLCHAIN_FILE="${CMAKE_TARGET_TOOLCHAIN}")
 CMAKE_FLAGS+=(-DCMAKE_BUILD_TYPE=Release -DEXAMPLES_ENABLE_C=OFF)
 CMAKE_FLAGS+=(-DKLU_ENABLE=ON -DKLU_INCLUDE_DIR="$prefix/include" -DKLU_LIBRARY_DIR="$libdir")
-CMAKE_FLAGS+=(-DBLAS_ENABLE=OFF -DLAPACK_ENABLE=OFF)
-
-#if [[ ${nbits} == 64 ]] && [[ ${target} != aarch64* ]]; then
-#    patch -p0 < $WORKSPACE/srcdir/patches/Sundials_ilp64.patch
-#    CMAKE_FLAGS+=(-DBLAS_LIBRARIES="${libdir}/libopenblas64_.${dlext}" -DLAPACK_LIBRARIES="${libdir}/libopenblas64_.${dlext}")
-#else
-#    CMAKE_FLAGS+=(-DBLAS_LIBRARIES="${libdir}/libopenblas.${dlext}" -DLAPACK_LIBRARIES="${libdir}/libopenblas.${dlext}")
-#fi
+CMAKE_FLAGS+=(-DLAPACK_ENABLE=ON)
+
+if [[ ${nbits} == 64 ]] && [[ ${target} != aarch64* ]]; then
+    # patch -p0 < $WORKSPACE/srcdir/patches/Sundials_fortran.patch
+    # export CFLAGS="-Wl,-rpath-link,/opt/${target}/${target}/sys-root/lib64"
+    CMAKE_FLAGS+=(-DLAPACK_LIBRARIES="${libdir}/libopenblas64_.${dlext}")
+else
+    CMAKE_FLAGS+=(-DLAPACK_LIBRARIES="${libdir}/libopenblas.${dlext}")
+fi
```

Comment:
This seems to work for me to enable support for LAPACK
```sh
if [[ ${nbits} == 64 ]] && [[ ${target} != aarch64* ]]; then
    atomic_patch -p1 $WORKSPACE/srcdir/patches/Sundials_Fortran.patch
    CMAKE_FLAGS+=(-DLAPACK_LIBRARIES="${libdir}/libopenblas64_.${dlext}")
else
    CMAKE_FLAGS+=(-DLAPACK_LIBRARIES="${libdir}/libopenblas.${dlext}")
fi
export CFLAGS="-lgfortran -lquadmath"
```

Reply:
I am locally getting this error when adding the CFLAGS export 

```
CMake Error at /usr/share/cmake/Modules/CMakeTestCCompiler.cmake:52 (message):
  The C compiler

    "/opt/bin/x86_64-linux-gnu-gcc"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: /workspace/srcdir/sundials-5.1.0/build/CMakeFiles/CMakeTmp
    
    Run Build Command:"/usr/bin/make" "cmTC_01425/fast"
    /usr/bin/make -f CMakeFiles/cmTC_01425.dir/build.make CMakeFiles/cmTC_01425.dir/build
    make[1]: Entering directory '/workspace/srcdir/sundials-5.1.0/build/CMakeFiles/CMakeTmp'
    Building C object CMakeFiles/cmTC_01425.dir/testCCompiler.c.o
    /opt/bin/x86_64-linux-gnu-gcc --sysroot=/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root/   -lgfortran -lquadmat    -o CMakeFiles/cmTC_01425.dir/testCCompiler.c.o   -c /workspace/srcdir/sundials-5.1.0/build/CMakeFiles/CMakeTmp/testCCompiler.c
    Linking C executable cmTC_01425
    /usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_01425.dir/link.txt --verbose=1
    /opt/bin/x86_64-linux-gnu-gcc --sysroot=/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root/ -lgfortran -lquadmat     CMakeFiles/cmTC_01425.dir/testCCompiler.c.o  -o cmTC_01425 
    /opt/x86_64-linux-gnu/bin/../lib/gcc/x86_64-linux-gnu/4.8.5/../../../../x86_64-linux-gnu/bin/ld: cannot find -lquadmat
    collect2: error: ld returned 1 exit status
    make[1]: *** [CMakeFiles/cmTC_01425.dir/build.make:87: cmTC_01425] Error 1
    make[1]: Leaving directory '/workspace/srcdir/sundials-5.1.0/build/CMakeFiles/CMakeTmp'
    make: *** [Makefile:121: cmTC_01425/fast] Error 2
    

  

  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:28 (project)
```

Reply:
Did you misspell `quadmath`?

Reply:
Dah... yeah 🤦‍♂ sorry